### ISSUE TITLE
Cross Validation breaking the temporal relationship

### ISSUE THREAD

**Issue by neto-riga** [LinkedInLearning/real-time-data-forecasting-with-ai-and-python-4565024#2]
In lesson 01_04 when using cross validation, aren't we breaking the temporal relationship our data? As my understand goes, we are splitting our data into 4 random splits. The danger is that future observations can end up in the training set while earlier observations are in the test set. To ensure that this does not happen I propose a few changes:

```
from sklearn.model_selection import TimeSeriesSplit
import xgboost as xgb
import numpy as np

params = {
    'eta': 0.05,
    'max_depth': 3,
    'min_child_weight': 5,
    'subsample': 0.8,
    'colsample_bytree': 0.8,
    'tree_method': 'hist',
    'verbosity': 2
}

tscv = TimeSeriesSplit(n_splits=4)
folds = list(tscv.split(X_train))   # returns (train_idx, test_idx) by POSITION

cv_results_ls = []
for h, y_train in enumerate(y_train_ls, start=1):
    dtrain = xgb.DMatrix(X_train.copy(), label=y_train)

    # Use the time-ordered folds; this overrides random nfold behavior
    cv_results = xgb.cv(
        params=params,
        dtrain=dtrain,
        folds=folds,                  # <-- key line
        num_boost_round=500,
        # early_stopping_rounds=30,
        metrics='rmse',
        as_pandas=True,
        seed=123
    )
    cv_results_ls.append(cv_results)
    print(f"H+{h} last mean RMSE: {cv_results['test-rmse-mean'].iloc[-1]}")
```

Had to change the params for stability, with default params it blowed.
Thanks for the content.
